test/odbc/psqlodbc/test_pg_datatype_bigint.cpp

```diff
@@ -0,0 +1,1016 @@
+  odbcHandler.ExecQuery(InsertStatement(TABLE_NAME, insert_string));
+
+  rcode = SQLRowCount(odbcHandler.GetStatementHandle(), &affected_rows);
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(affected_rows, NUM_OF_INSERTS);
+
+  odbcHandler.CloseStmt();
+
+  // Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string>{COL1_NAME}));
+
+  // Make sure inserted values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(BIND_COLUMNS));
+
+  for (int i = 0; i < NUM_OF_INSERTS; i++) {
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+    ASSERT_EQ(pk_len, PK_BYTES_EXPECTED);
+    ASSERT_EQ(pk, i);
+    if (INSERTED_VALUES[i] != "NULL") {
+      ASSERT_EQ(data_len, DATA_BYTES_EXPECTED);
+      ASSERT_EQ(data, StringToBigInt(INSERTED_VALUES[i]));
+    }
+    else {
+      ASSERT_EQ(data_len, SQL_NULL_DATA);
+    }
+  }
+
+  // Assert that there is no more data
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_BigInt, Insertion_Fail) {
+  RETCODE rcode;
+  OdbcHandler odbcHandler(Drivers::GetDriver(ServerType::PSQL));
+
+  const vector<string> INVALID_INSERTED_VALUES = {
+    "-9223372036854775809",
+    "9223372036854775808"
+  };
+  const int NUM_OF_INSERTS = INVALID_INSERTED_VALUES.size();
+
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS));
+  odbcHandler.CloseStmt();
+
+  // Attempt to insert values that are out of range and assert that they all fail
+  for (int i = 0; i < NUM_OF_INSERTS; i++) {
+    string insert_string = "(" + std::to_string(i) + "," + INVALID_INSERTED_VALUES[i] + ")";
+
+    rcode = SQLExecDirect(odbcHandler.GetStatementHandle(), (SQLCHAR *)InsertStatement(TABLE_NAME, insert_string).c_str(), SQL_NTS);
+    ASSERT_EQ(rcode, SQL_ERROR);
+  }
+
+  // Select all from the tables and assert that nothing was inserted
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string>{COL1_NAME}));
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_BigInt, Update_Success) {
+  const int PK_INSERTED = 1;
+  const string DATA_INSERTED = "1";
+
+  const vector<string> DATA_UPDATED_VALUES = {
+    "-9223372036854775808",
+    "9223372036854775807",
+    "123456789"
+  };
+  const int NUM_OF_DATA = DATA_UPDATED_VALUES.size();
+
+  const string INSERT_STRING = "(" + std::to_string(PK_INSERTED) + "," + DATA_INSERTED + ")";
+  const string UPDATE_WHERE_CLAUSE = COL1_NAME + " = " + std::to_string(PK_INSERTED);
+
+  const int PK_BYTES_EXPECTED = 4;
+  const int DATA_BYTES_EXPECTED = 8;  
```

Comment:
`PK_BYTES_EXPECTED` and `DATA_BYTES_EXPECTED` are used in several locations. Should we consider to make this global? Or maybe a more general case and name it `BIGINT_EXPECTED_BYTES` and `INT_EXPECTED_BYTES`?